### PR TITLE
Don’t add format to legends if we use bin because we have a special f…

### PR DESCRIPTION
**Pull request by domoritz**
…ormat that displays the range.

### CODE REVIEW

**Comment by kanitw** [comment]
The change LGTM. 

But I prefer would like to have `model.format()` either
- return `def` so it's clear that the method cause side effect on the input `def`
- or do something like

```js
extend(def, model.format()); 
```
where `model.format()` only returns mix-ins part of the format.   (And thus, possibly rename it to `model.formatMixins()`.)    

The latter is consistent with `compile/util.ts`'s `applyColorMixins` and scale's `rangeMixins`.  

This is slightly slower, but is clearer that we are modifying def.

**Comment by domoritz** [comment]
> This is slightly slower, but is clearer that we are modifying def. 

Performance doesn't matter for small things like this, imho. I'll change this.

**Comment by kanitw** [comment]
Cool.  And maybe unit test? :p

**Comment by domoritz** [comment]
> return def so it's clear that the method cause side effect on the input def

It's not clear because you'd modify and return the def. Making a copy is so painful is js. 

> The latter is consistent with `compile/util.ts`'s `applyColorMixins` and scale's `rangeMixins`.

`compile/util.ts` doesn't have`applyColorMixins`. And `applyColorAndOpacity` actually writes directly.

**Comment by kanitw** [comment]
I didn’t check that when I write lol … it’s purely from my bad memory.  

Thinking more about what I did, I think with the name `applyColorAndOpacity()` then it makes sense not to return `mixin`.  (Otherwise it should be named `colorAndOpacityMixins()`.)

> scale's `rangeMixins`.
This is still true though.

**Comment by domoritz** [comment]
Agreed. We can easily make this a mixing, though by not taking the p parameter and just returning and incomplete object.

**Comment by kanitw** [comment]
> We can easily make this a mixing, though by not taking the p parameter and just returning and incomplete object.

Yeah, but given that `applyColorAndOpacity()` is in `util.ts`, not `model.ts`.  The original `apply` method already make sense IMHO.

**Comment by domoritz** [comment]
Done.

File: src/compile/legend.ts [hunk 1]
```diff
@@ -71,6 +71,18 @@ export function title(fieldDef: FieldDef) {
...
+export function format(model: Model, channel: Channel) {
+  const fieldDef = model.fieldDef(channel);
+
+  // If the channel is binned, we should not set the formatter because we have a range label
+  if (!fieldDef.bin) {
+    return {};
+  }
+
+  const legend = fieldDef.legend;
+  return model.format(channel, (typeof legend !== 'boolean' && legend.format) || undefined);
```
Comment (kanitw) [inline src/compile/legend.ts]: should we introduce `model.legend()` here just like how we have `model.axis()` for consistency?

**Comment by kanitw** [comment]
Don’t we have to apply `model.format()` to `mark-text.ts` as well?

**Pull request merged**